src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -743,7 +754,7 @@ public void flush(FlushType type, boolean force, boolean waitIfOngoing) throws E
                         { // commit and close the current writer - we write the current tanslog ID just in case
                             final long translogId = translog.currentId();
                             indexWriter.setCommitData(Collections.singletonMap(Translog.TRANSLOG_ID_KEY, Long.toString(translogId)));
-                            indexWriter.commit();
+                            commitIndexWriter(indexWriter);
```

Comment:
downgrading a writeLock is fine - the problems is different that some caller holds the read lock and we fail the engine wich in-turn closes it (and acquires a write lock)